src/lib.rs

```diff
@@ -1,110 +1,451 @@
-use std::fmt::Write;
+//! A document in the ROFF format.
+//!
+//! [ROFF] is a family of Unix text-formatting languages, implemented
+//! by the `nroff`, `troff`, and `groff` programs, among others. See
+//! [groff(7)] for a description of the language. This structure is an
+//! abstract representation of a document in ROFF format. It is meant
+//! for writing code to generate ROFF documents, such as manual pages.
+//!
+//! # Example
+//!
+//! ```
+//! # use roff::*;
+//! let doc = Builder::default().text(vec![roman("hello, world")]).build().render();
+//! assert!(doc.ends_with("hello, world\n"));
+//! ```
+//!
+//! [ROFF]: https://en.wikipedia.org/wiki/Roff_(software)
+//! [groff(7)]: https://manpages.debian.org/bullseye/groff/groff.7.en.html
 
-#[derive(PartialEq, Eq)]
+#![deny(missing_docs)]
+
+use std::io::Write;
+use std::write;
+
+// A preamble added to the start of rendered output.
```

Comment:
nit: There a reason this is only a comment and not a doc-comment?  I know its internal but seems like it'd be nice to have it at least highlighted by editors as a doc comment (unsure how many people run rustdoc for private items, unsure about IDE integration benefits)